### DISCUSSION TITLE
.XML FILE Fast DDS

### DISCUSSION THREAD

**Discussion by AndresGF8** [eProsima/Fast-DDS#4529] [Q&A]
Hi, I’ve been trying for a few weeks to connect the Matlab-Simulink software with some subscriber nodes and publishers via FastDS, making use of the DDS Blockset block.
To do this, I need to create an . XML file in which the domain, themes, and subscriber/publisher nodes I want to create are declared. As much as I try to look for examples, I cannot create a correct code, since Simulink never recognizes the topics when I assign it to the DDS Blockset. The code is as follows:

<?xml version="1.0" encoding="UTF-8"?>
<dds xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">

    <types>
      <struct name="Header">
        <member name="data" type="int32"/>
      </struct>
    </types>

    <domainLibrary profile_name="domainLibrary_profile">

    <profiles>
      <participant profile_name="participant_xml_profile">
        <rtps>
          <name>my_participant</name>
          <builtin>
            <domainId>0</domainId>
          </builtin>
        </rtps>
      </participant>

      <publisher profile_name="PublicoEnJuan">
        <topic>
          <name>juan</name>
          <dataType>Header</dataType>
        </topic>
      </publisher>

      <subscriber profile_name="LeoEnAndres">
        <topic>
          <name>andres</name>
          <dataType>Header</dataType>
        </topic>
      </subscriber>

    </profiles>
  </domainLibrary>
</dds>

As I mentioned earlier, the Simulink block does detect for example the type "Header" that I have declared, but it does not recognize either the domain library, the specified domain or the topics within that domain. The only thing I want to implement is a file .XML that works in such a way that a subscriber node (andres) subscribes to a topic by which integers arrive and publish them through a publishing node (juan) on a different topic.

A greeting and thanks in advance

**Comment by JesusPoderoso** [comment]
Hi @AndresGF8, thanks for using Fast DDS.
I am afraid this question should be addressed by the Matlab-Simulink DDS Blockset software maintainers.
We are not distributing Fast DDS over Matlab/Simulink, so we cannot help you with that configuration.
Happy to help with any other questions regarding Fast DDS in our [supported platforms](https://fast-dds.docs.eprosima.com/en/latest/notes/versions.html#build-system-dependencies)!
Sorry for the inconvenience.

**Reply by JLBuenoLopez** [reply]
@AndresGF8 this is not the proper forum for your question. The Matlab/Simulink configuration for Fast DDS is something not depending on this repository. I am afraid that we are unable to help you with your specific configuration. Consider writing to the Matlab/Simulink developers or try finding something in their documentation.

**Comment by MiguelCompany** [comment]
Something you can try, though, is checking your XML against our [XSD schema](https://github.com/eProsima/Fast-DDS/blob/master/resources/xsd/fastRTPS_profiles.xsd)

Looking at the XML you posted, I see that you should remove line `<domainLibrary profile_name="domainLibrary_profile">`.

I would also have the `profiles` tag before the `types` one.

**Reply by AndresGF8** [reply]
Hello, thanks for the answer. 
As you suggested, I also tried to use your "XSD schema" but I get a major error, since the Simulink block does not let me enter the file .XML because it says it doesn’t recognize startup tags. With my code option at least it recognizes the data type so I don't really know what I can do. 

Thank you again

**Comment by AndresGF8** [comment]
> Hi, I’ve been trying for a few weeks to connect the Matlab-Simulink software with some subscriber nodes and publishers via FastDS, making use of the DDS Blockset block. To do this, I need to create an . XML file in which the domain, themes, and subscriber/publisher nodes I want to create are declared. As much as I try to look for examples, I cannot create a correct code, since Simulink never recognizes the topics when I assign it to the DDS Blockset. The code is as follows:
> 
> ```
> <types>
>   <struct name="Header">
>     <member name="data" type="int32"/>
>   </struct>
> </types>
> 
> <domainLibrary profile_name="domainLibrary_profile">
> 
> <profiles>
>   <participant profile_name="participant_xml_profile">
>     <rtps>
>       <name>my_participant</name>
>       <builtin>
>         <domainId>0</domainId>
>       </builtin>
>     </rtps>
>   </participant>
> 
>   <publisher profile_name="PublicoEnJuan">
>     <topic>
>       <name>juan</name>
>       <dataType>Header</dataType>
>     </topic>
>   </publisher>
> 
>   <subscriber profile_name="LeoEnAndres">
>     <topic>
>       <name>andres</name>
>       <dataType>Header</dataType>
>     </topic>
>   </subscriber>
> 
> </profiles>
> ```
> 
> As I mentioned earlier, the Simulink block does detect for example the type "Header" that I have declared, but it does not recognize either the domain library, the specified domain or the topics within that domain. The only thing I want to implement is a file .XML that works in such a way that a subscriber node (andres) subscribes to a topic by which integers arrive and publish them through a publishing node (juan) on a different topic.
> 
> A greeting and thanks in advance

I've just realized I've not posted the complete code:


<?xml version="1.0" encoding="UTF-8"?>
<dds xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">


    <types>
      <struct name="Header">
        <member name="data" type="int32"/>
      </struct>
    </types>

    <domainLibrary profile_name="domainLibrary_profile">

    <profiles>
      <participant profile_name="participant_xml_profile">
        <rtps>
          <name>my_participant</name>
          <builtin>
            <domainId>0</domainId>
          </builtin>
        </rtps>
      </participant>

      <publisher profile_name="PublicoEnJuan">
        <topic>
          <name>juan</name>
          <dataType>Header</dataType>
        </topic>
      </publisher>

      <subscriber profile_name="LeoEnAndres">
        <topic>
          <name>andres</name>
          <dataType>Header</dataType>
        </topic>
      </subscriber>

    </profiles>
  </domainLibrary>
</dds>
